src/services/PhoneHome.php

```diff
@@ -0,0 +1,167 @@
+
+        $existingRecord = self::_existingRecord();
+
+        try {
+            if ($existingRecord === null) {
+                $request = self::$_airtable->saveContent(self::$_tableName, self::_appInfo());
+            } else {
+                $request = self::$_airtable->updateContent(self::$_tableName, [
+                    [
+                        'id' => $existingRecord,
+                        'fields' => self::_appInfo(),
+                    ],
+                ]);
+            }
+        } catch (\Exception $e) {
+            Craft::error(
+                'Error phoning home',
+                __METHOD__
+            );
+        }
+
+        $oneDayDuration = 60 * 60 * 24;
+        Craft::$app->cache->set(self::$_cacheKey, true, $oneDayDuration);
+    }
+
+    /**
+     * Return the existing record's ID (if one exists)
+     *
+     * @return string|null
+     */
+    private static function _existingRecord(): ?string
+    {
+        $request = self::$_airtable->quickCheck(self::$_tableName, 'Site URL', self::$_siteUrl);
+
+        // No existing record
+        if ($request->count === 0) return null;
+
+        return $request->records['0']->id ?? null;
+    }
+
+    /**
+     * Returns all the info we want to send to Airtable
+     *
+     * @return array
+     */
+    private static function _appInfo(): array
+    {
+        return [
+            'Site URL' => self::$_siteUrl,
+            'Site Name' => Craft::$app->getSystemName(),
+            'Craft Version' => 'Craft ' . App::editionName(Craft::$app->getEdition()) . ' ' . Craft::$app->getVersion(),
+            'PHP Version' => App::phpVersion(),
+            'DB Version' => self::_dbDriver(),
+            'Plugins' => self::_plugins(),
+            'Modules' => self::_modules(),
+        ];
+    }
+
+    /**
+     * Returns the DB driver name and version
+     *
+     * @return string
+     */
+    private static function _dbDriver(): string
+    {
+        $db = Craft::$app->getDb();
+
+        if ($db->getIsMysql()) {
+            $driverName = 'MySQL';
+        } else {
+            $driverName = 'PostgreSQL';
+        }
+
+        return $driverName . ' ' . App::normalizeVersion($db->getSchema()->getServerVersion());
+    }
+
+    /**
+     * Returns the list of plugins and versions
+     *
+     * @return string
+     */
+    private static function _plugins(): string
+    {
+        $plugins = Craft::$app->plugins->getAllPlugins();
+
+        return implode(PHP_EOL, array_map(function($plugin) {
+            return "{$plugin->name} ({$plugin->developer}): {$plugin->version}";
+        }, $plugins));
+    }
+
+    /**
+     * Returns the list of modules
+     *
+     * @return string
+     */
+    private static function _modules(): string
+    {
+        $modules = [];
+
+        foreach (Craft::$app->getModules() as $id => $module) {
+            if ($module instanceof PluginInterface) {
```

Comment:
plugins show up here as well?

Reply:
Yeah, because technically they are modules under the hood (with a little extra on top). I ripped this stuff straight from Craft for how they generated their system report page

Reply:
smart